### PR TITLE
gh-107659: Improve wording of the description of `ctypes.pointer` and `ctypes.POINTER`

### PR DESCRIPTION
**Pull request by tomasr8**
In a [previous PR](https://github.com/python/cpython/pull/107660), docstrings for these two methods were added. The description from the docs was used for the docstrings with some small improvements. It was discussed that we could apply these improvements to the docs as well so that they stay in sync: https://github.com/python/cpython/pull/107660#issuecomment-1666810437

The change is mostly to use the imperative mode when describing the functions and dropping some redundant words. I also updated the function signature.

I think neither an issue or a NEWS entry is required since this is a small change, but let me know if that's not the case :)

----
:books: Documentation preview :books:: https://cpython-previews--107769.org.readthedocs.build/

* Issue: gh-107659

### CODE REVIEW

File: Doc/library/ctypes.rst [hunk 1]
```diff
@@ -2029,18 +2029,18 @@ Utility functions
...
 
-.. function:: POINTER(type)
+.. function:: POINTER(type, /)
 
-   This factory function creates and returns a new ctypes pointer type. Pointer
-   types are cached and reused internally, so calling this function repeatedly is
-   cheap. *type* must be a ctypes type.
+   Create and return a new ctypes pointer type. Pointer types are cached and
+   reused internally, so calling this function repeatedly is cheap. *type*
+   must be a ctypes type.
```
Comment (AA-Turner) [inline Doc/library/ctypes.rst]: Minor, to have the sentence start on a new line
```suggestion
   reused internally, so calling this function repeatedly is cheap.
   *type* must be a ctypes type.
```

**Review by AA-Turner — APPROVED** [review]
Thank you!

A

File: Doc/library/ctypes.rst [hunk 2]
```diff
@@ -2029,18 +2029,18 @@ Utility functions
...
 
-.. function:: pointer(obj)
-
-   This function creates a new pointer instance, pointing to *obj*. The returned
-   object is of the type ``POINTER(type(obj))``.
+.. function:: pointer(obj, /)
 
+   Create a new pointer instance, pointing to *obj*.
+   The returned object is of the type ``POINTER(type(obj))``.
+
```
Comment (AA-Turner) [inline Doc/library/ctypes.rst]:
```suggestion

```

File: Doc/library/ctypes.rst [hunk 3]
```diff
@@ -2029,18 +2029,17 @@ Utility functions
...
 
 
-.. function:: pointer(obj)
-
-   This function creates a new pointer instance, pointing to *obj*. The returned
-   object is of the type ``POINTER(type(obj))``.
+.. function:: pointer(obj, /)
 
+   Create a new pointer instance, pointing to *obj*.
+   The returned object is of the type ``POINTER(type(obj))``.
```
Comment (AA-Turner) [inline Doc/library/ctypes.rst:2042]:
```suggestion
   The returned object is of the type ``POINTER(type(obj))``.

```

**Review by kumaraditya303 — APPROVED** [review]

**Pull request merged**